indico/web/templates/forms/multiple_items_widget.html

```diff
@@ -152,20 +181,92 @@
             }
 
             function makeColData(item, col, forceEditable) {
-                return (item && !forceEditable) ? {
-                    text: item[col[0]]
-                } : {
-                    html: $('<input>', {
-                        'type': 'text',
-                        'class': 'js-table-input',
-                        'value': item ? item[col[0]] : ''
-                        // can't set placeholder here due to the jquery.placeholder plugin
-                    }).attr('placeholder', col[1])
+                if (item && !forceEditable) {
+                    return col.type === 'select' ? {text: columnChoices[col.id][item[col.id]]} : {text: item[col.id]};
+                } else {
+                    if (col.type === 'select') {
+                        var sel = $('<select>', {
+                            'class': 'js-table-input',
+                            'data-required': col.required
+                        });
+                        sel.append($('<option>')); // Default empty option
+                        for (var choiceID in columnChoices[col.id]) {
+                            sel.append($('<option>', {
+                                value: choiceID,
+                                text: columnChoices[col.id][choiceID],
+                                selected: item ? item[col.id] === choiceID : false
+                            }));
+                        }
+                        return {html: sel};
+                    } else if(col.type === 'number') {
+                        var numberInput = $('<input>', {
+                            'type': 'number',
+                            'class': 'js-table-input',
+                            'value': item ? item[col.id] : '',
+                            'placeholder': col.caption,
+                            'data-required': col.required,
+                            'min': col.min,
+                            'max': col.max,
+                            'step': 'any' // To accept decimal value
```

Comment:
can we take this from the config dict? for the fees, a 0.01 step would be useful for example. anyway, maybe overkill for now..